saleor/graphql/attribute/types.py

```diff
@@ -656,3 +655,791 @@ class AttributeValueInput(BaseInputObjectType):
+        ),
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = "Represents multi collection reference attribute." + ADDED_IN_322
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_value(
+        root: AssignedAttributeData,
+        info: ResolveInfo,
+        limit: int = DEFAULT_NESTED_LIST_LIMIT,
+    ) -> Promise[list[ChannelContext[product_models.Collection]]]:
+        if not root.values:
+            return Promise.resolve([])
+
+        channel_slug = root.attribute.channel_slug
+        attr_values = [value.node for value in root.values]
+        attr_values = attr_values[:limit]
+
+        def _wrap_with_channel_context(
+            collections: list[product_models.Collection],
+        ) -> list[ChannelContext[product_models.Collection]]:
+            if not collections:
+                return []
+            return [
+                ChannelContext(node=collection, channel_slug=channel_slug)
+                for collection in collections
+            ]
+
+        return (
+            CollectionByIdLoader(info.context)
+            .load_many([value.reference_collection_id for value in attr_values])
+            .then(_wrap_with_channel_context)
+        )
+
+
+class AssignedChoiceAttributeValue(BaseObjectType):
+    name = graphene.String(description=AttributeValueDescriptions.NAME)
+    slug = graphene.String(description=AttributeValueDescriptions.SLUG)
+    translation = graphene.String(
+        language_code=graphene.Argument(
+            LanguageCodeEnum,
+            required=True,
+        ),
+        description="Translation of the name.",
+        required=False,
+    )
+
+    class Meta:
+        description = (
+            "Represents a single choice value of the attribute." + ADDED_IN_322
+        )
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_translation(
+        root: AttributeValue, info: ResolveInfo, *, language_code
+    ) -> Promise[str | None] | None:
+        return (
+            AttributeValueTranslationByIdAndLanguageCodeLoader(info.context)
+            .load((root.id, language_code))
+            .then(lambda translation: translation.name if translation else None)
+        )
+
+
+class AssignedSingleChoiceAttribute(BaseObjectType):
+    value = graphene.Field(
+        AssignedChoiceAttributeValue,
+        required=False,
+        description="The assigned choice value.",
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = "Represents a single choice attribute." + ADDED_IN_322
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    def resolve_value(
+        root: AssignedAttributeData, info: ResolveInfo
+    ) -> models.AttributeValue | None:
+        if not root.values:
+            return None
+        attr_value = root.values[0].node
+        return attr_value
+
+
+class AssignedMultiChoiceAttribute(BaseObjectType):
+    value = NonNullList(
```

Comment:
In this field, Saleor returns multiple values. 
```suggestion
    values = NonNullList(
```